File: CSHARP/Cap07/c07ex09.cs
// c07ex09.cs

using System;
using static System.Console;

class Program
{
    static int FatorialBase(int N, int P)
    {
        if (N == 0)
            return P;
        else
            return FatorialBase(N - 1, N * P);
    }

    static int Fatorial(int N)
    {
        return FatorialBase(N, 1);
    }

    public static void Main(string[] args)
    {
        int Limite;

        Write("Qual fatorial: ");
        Limite = int.Parse(ReadLine());

        WriteLine($"Fatorial = {Fatorial(Limite)}");

        Write("\nTecle <Enter> para encerrar... ");
        ReadLine();
    }
}

File: DART/Cap07/c07ex09.dart
// c07ex09.dart

import 'dart:io';

int fatorialBase(int N, int P) {
  if (N == 0)
    return P;
  else
    return fatorialBase(N - 1, N * P);
}

int fatorial(int N) {
  return fatorialBase(N, 1);
}

void main() {
  int limite;

  stdout.write("Qual fatorial: ");
  limite = int.tryParse(stdin.readLineSync() ?? '') ?? 0;

  print("Fatorial = ${fatorial(limite)}");

  stdout.write("\nTecle <Enter> para encerrar... ");
  stdin.readLineSync();
}
